woocommerce/woocommerce-android
# Fix deleting product

Part of: #9541 

### Why
When deleting a bundle product, we must ensure that grouped products are deleted in conjunction with its children's products

### Description
This PR changes how we delete products from the order in the order creation flow. Instead of only sending the list item information to remove the list item from the order, we send the order creation product so we can properly handle grouped product deletion.

### Testing instructions

1. Open the order tab
2. Tap on create new order
3. Select a bundle product
4. Tap on the bundle product and navigate to the product details screen
5. Tap on Remove product from order and check that the order contains no products (the bundle product is deleted with its children products)

### Images/gif

#### Before

https://github.com/woocommerce/woocommerce-android/assets/18119390/48994953-1fb0-4453-b946-a8fb61e4db32

#### After

https://github.com/woocommerce/woocommerce-android/assets/18119390/44202053-31ba-489f-8808-3a83f35775dc
